### DISCUSSION TITLE
OneSignal-vue and Quasar

### DISCUSSION THREAD

**Discussion by IvanWala** [quasarframework/quasar#11714] [General - Components / Directives / etc]
Hello I need to install onesignal in a vue application. 

https://www.npmjs.com/package/onesignal-vue
https://github.com/OneSignal/onesignal-vue

I start to install this plugin in a boot file:

```
import { boot } from 'quasar/wrappers'
import OneSignalVue from 'onesignal-vue'

export default boot(({ app }) => {
    app.use(OneSignalVue);
});
```
But now I don't have idea where to place this:

```
new Vue({
  render: h => h(App),
  beforeMount() {
    this.$OneSignal.init({ appId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' });
  }
}).$mount('#app')
```

Some one can help me please?

Thanks for Quasar.

**Comment by hawkeye64** [comment]
@IvanWala Add it directly to your App.vue file, located in the `src` folder:

```js
<template>
  <router-view />
</template>

<script>
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'App',
  beforeMount () {
    this.$OneSignal.init({ appId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' });
  }
})
</script>
```

**Comment by IvanWala** [comment]
Thank you @hawkeye64. 

In this moment it throws an error. And probably is because I need to integrate the OneSignal Service worker. When they say

https://github.com/OneSignal/onesignal-vue
https://documentation.onesignal.com/docs/web-push-quickstart#step-6-upload-files

I can make this?:

For pwa mode, edit the generated src-pwa/register-service-worker.js and change
`register(process.env.SERVICE_WORKER_FILE, {`
to
`register('/OneSignalSDKWorker.js', {`

like here

https://www.npmjs.com/package/quasar-app-extension-onesignal

PD: I try to use this plugin before but it doesn't works.

**Comment by IvanWala** [comment]
I integrate the service worker as "quasar-app-extension-onesignal" plugin says, i put the appId and it throws this error.

```
vendor.fc9630ff.js:8 TypeError: Cannot read properties of undefined (reading 'init')
    at Proxy.beforeMount (app.3be19be0.js:1)
    at ln (vendor.fc9630ff.js:8)
    at un (vendor.fc9630ff.js:8)
    at Array.N.t.__weh.t.__weh (vendor.fc9630ff.js:8)
    at Q (vendor.fc9630ff.js:8)
    at w.u [as fn] (vendor.fc9630ff.js:8)
    at w.run (vendor.fc9630ff.js:8)
    at j (vendor.fc9630ff.js:8)
    at P (vendor.fc9630ff.js:8)
    at R (vendor.fc9630ff.js:8)
fn @ vendor.fc9630ff.js:8
cn @ vendor.fc9630ff.js:8
(anonymous) @ vendor.fc9630ff.js:8
Promise.catch (async)
un @ vendor.fc9630ff.js:8
N.t.__weh.t.__weh @ vendor.fc9630ff.js:8
Q @ vendor.fc9630ff.js:8
u @ vendor.fc9630ff.js:8
run @ vendor.fc9630ff.js:8
j @ vendor.fc9630ff.js:8
P @ vendor.fc9630ff.js:8
R @ vendor.fc9630ff.js:8
w @ vendor.fc9630ff.js:8
X @ vendor.fc9630ff.js:8
mount @ vendor.fc9630ff.js:8
t.mount @ vendor.fc9630ff.js:8
_ @ app.3be19be0.js:1
await in _ (async)
(anonymous) @ app.3be19be0.js:1
Promise.then (async)
(anonymous) @ app.3be19be0.js:1
Promise.then (async)
9757 @ app.3be19be0.js:1
r @ app.3be19be0.js:1
(anonymous) @ app.3be19be0.js:1
r.O @ app.3be19be0.js:1
(anonymous) @ app.3be19be0.js:1
(anonymous) @ app.3be19be0.js:1
A bad HTTP response code (404) was received when fetching the script.
```

I

**Reply by hawkeye64** [reply]
It's a 404. It can't find your script. Did you put it into the public folder?

**Comment by IvanWala** [comment]
Hi and thanks again @haweye64. This is the link of a "beta" app.

https://app.walaxia.cat/

**Reply by hawkeye64** [reply]
It's getting an error here:

```js
          await this.$OneSignal.init({
            appId: "7919112d-9e96-47e1-a054-b2218842f425"
          })
```

This doesn't appear to be hooked up properly: `this.$OneSignal`

**Comment by IvanWala** [comment]
Hi again @hawkeye64. And how can manage 

`this.$OneSignal `

in beforeEach with this in a boot?

```
import { boot } from 'quasar/wrappers'
import OneSignalVue from 'onesignal-vue'

export default boot(({ app }) => {
    app.use(OneSignalVue);
});
```

**Comment by hawkeye64** [comment]
I haven't looked at the OneSignalVue code, so I can only presume...

boot file
```js
import { boot } from 'quasar/wrappers'
import OneSignalVue from 'onesignal-vue'

export default boot(({ app }) => {
    app.use(OneSignalVue);
});

export OneSignalVue;
```

Other file:
```js
import { OneSignalVue } from 'boot/<boot name>'
await OneSignalVue.init({
  appId: "7919112d-9e96-47e1-a054-b2218842f425"
})
```

**Comment by IvanWala** [comment]
Hi again @hawkeye64 . 

onesignal-vue inject to the app.prototype but doesn't export OneSignalVue. That's the end of the index.

```
var OneSignalVuePlugin = {
    install: function (app) {
        app.prototype.$OneSignal = OneSignalVue;
    }
};

module.exports = OneSignalVuePlugin;
```


https://github.com/OneSignal/onesignal-vue/issues/4

**Comment by IvanWala** [comment]
I connect with OneSignal Discord and they doesn't has ready their plugin for vue3 and Quasar 2 in a short time.

**Comment by hawkeye64** [comment]
@IvanWala That makes sense. A lot of 3rd-party libraries are still playing catch-up, even 1 year later.
